serverless/src/shared/getFilteredTriples.js

```diff
@@ -1,70 +1,109 @@
+ * // Fetch triples for a specific concept scheme in the published version, page 1 with 10 items per page
  * try {
- *   const triples = await getFilteredTriples({ conceptScheme: 'sciencekeywords', version: 'published' });
+ *   const triples = await getFilteredTriples({
+ *     conceptScheme: 'sciencekeywords',
+ *     version: 'published',
+ *     pageNum: 1,
+ *     pageSize: 10
+ *   });
  *   console.log(triples);
  * } catch (error) {
  *   console.error('Failed to get triples:', error);
  * }
  *
  * @example
- * // Fetch triples matching a pattern in the draft version
+ * // Fetch triples matching a pattern in the draft version, page 2 with 20 items per page
  * try {
- *   const triples = await getFilteredTriples({ pattern: 'EARTH SCIENCE', version: 'draft' });
+ *   const triples = await getFilteredTriples({
+ *     pattern: 'EARTH SCIENCE',
+ *     version: 'draft',
+ *     pageNum: 2,
+ *     pageSize: 20
+ *   });
  *   console.log(triples);
  * } catch (error) {
  *   console.error('Failed to get triples:', error);
  * }
  *
- * @see getTriplesForConceptSchemeOrPatternQuery - Used to construct the SPARQL query.
- * @see sparqlRequest - Used to make the SPARQL query request.
+ * @see sparqlRequest - Used to make the SPARQL query requests.
+ * @see getConceptUrisQuery - Used to generate the SPARQL query for fetching concept URIs.
+ * @see getConceptDetailsQuery - Used to generate the SPARQL query for fetching concept details.
  */
 
-export const getFilteredTriples = async ({ conceptScheme, pattern, version }) => {
+export const getFilteredTriples = async ({
+  conceptScheme, pattern, version, pageNum, pageSize
+}) => {
+  const offset = (pageNum - 1) * pageSize
+
   try {
-    const response = await sparqlRequest({
+    // Step 1: Fetch concept URIs
+    const uriQuery = getConceptUrisQuery({
+      conceptScheme,
+      pattern,
+      pageSize,
+      offset
+    })
+    const uriResponse = await sparqlRequest({
+      method: 'POST',
+      contentType: 'application/sparql-query',
+      accept: 'application/sparql-results+json',
+      body: uriQuery,
+      version
+    })
+
+    if (!uriResponse.ok) {
+      throw new Error(`HTTP error! status: ${uriResponse.status}`)
+    }
+
+    const uriJson = await uriResponse.json()
+    const conceptUris = uriJson.results.bindings.map((binding) => binding.s.value)
+
+    // Step 2: Fetch details for these concepts
+    const query = getConceptDetailsQuery(conceptUris)
+    const detailsResponse = await sparqlRequest({
       method: 'POST',
       contentType: 'application/sparql-query',
       accept: 'application/sparql-results+json',
-      body: getTriplesForConceptSchemeOrPatternQuery({
-        conceptScheme,
-        pattern
-      }),
+      body: query,
       version
     })
 
-    if (!response.ok) {
-      throw new Error(`HTTP error! status: ${response.status}`)
+    if (!detailsResponse.ok) {
+      throw new Error(`HTTP error! status: ${detailsResponse.status}`)
```

Comment:
```suggestion
      throw new Error(`HTTP error! status: ${detailsResponse.status}`)
```
Same with this one. I imagine that this is pretty consistent in the project maybe in the future we can go back and improve the logs a bit to make it more obvious where the error was happening. In the scope of this effort either is fine with me

Reply:
done